java/src/main/java/com/saucelabs/saucebindings/configs/SauceConfigsBrowser.java

```diff
@@ -0,0 +1,127 @@
+package com.saucelabs.saucebindings.configs;
+
+import com.saucelabs.saucebindings.PageLoadStrategy;
+import com.saucelabs.saucebindings.Prerun;
+import com.saucelabs.saucebindings.UnhandledPromptBehavior;
+import org.openqa.selenium.Proxy;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.Map;
+
+public class SauceConfigsBrowser<T extends SauceConfigsBrowser<T>> extends SauceConfigs<T> {
+
+    // W3C Values
+
+    // TODO: Set restrictions on allowed versions in subclass
+    public T setBrowserVersion(String browserVersion) {
+        sauceOptions.setBrowserVersion(browserVersion);
+        return (T) this;
+    }
+
+    public T setPageLoadStrategy(PageLoadStrategy pageLoadStrategy) {
+        sauceOptions.setPageLoadStrategy(pageLoadStrategy);
+        return (T) this;
+    }
+
+    public T setAcceptInsecureCerts(Boolean bool) {
+        sauceOptions.setAcceptInsecureCerts(bool);
+        return (T) this;
+    }
+
+    public T setProxy(Proxy proxy) {
+        sauceOptions.setProxy(proxy);
+        return (T) this;
+    }
+
+    public T setSetWindowRect(Boolean bool) {
+        sauceOptions.setSetWindowRect(bool);
+        return (T) this;
+    }
+
+    public T setStrictFileInteractability(Boolean bool) {
+        sauceOptions.setStrictFileInteractability(bool);
+        return (T) this;
+    }
+
+    public T setUnhandledPromptBehavior(UnhandledPromptBehavior behavior) {
+        sauceOptions.setUnhandledPromptBehavior(behavior);
+        return (T) this;
+    }
+
+    // Sauce Values common to all browser sessions:
+    public T setRecordVideo(Boolean bool) {
+        sauceOptions.setRecordVideo(bool);
+        return (T) this;
+    }
+
+
+    public T setVideoUploadOnPass(Boolean bool) {
```

Comment:
I'm not sure if this is the time to do it, but I've been wondering if it makes sense to be passing Booleans in our setters. All of these capabilities are either enabled or disabled by default. So our methods only really do one of the options. . Does it make sense for a user to do `setVideoUploadOnPass(true)` when the only reasonable option is `disableVideoUploadOnPass()`?

Reply:
Yes! I like this! Passing in Boolean parameters has always seemed clunky to me.

Reply:
Ok, updated the code to use `disable` when the default is true and `set` when the default is false; neither takes an argument